src/smart-components/role/add-role-new/review-step-buttons.js

```diff
@@ -0,0 +1,83 @@
+import { Button } from '@patternfly/react-core';
+import React, { Fragment, useContext } from 'react';
+import PropTypes from 'prop-types';
+import get from 'lodash/get';
+import { AddRoleWizardContext } from './add-role-wizard';
+import useFormApi from '@data-driven-forms/react-form-renderer/dist/cjs/use-form-api';
+
+const selectNext = (nextStep, getState) => {
+  if (typeof nextStep === 'string') {
+    return nextStep;
+  }
+
+  if (typeof nextStep === 'function') {
+    return nextStep({ values: getState().values });
+  }
+
+  const selectedValue = get(getState().values, nextStep.when);
+
+  return nextStep.stepMapper[selectedValue];
+};
+
+const NextButton = ({ nextStep, valid, handleNext, nextLabel, getState, handleSubmit, submitLabel }) => (
+  <Button
+    variant="primary"
+    type="button"
+    isDisabled={!valid || getState().validating}
+    onClick={() => (nextStep ? handleNext(selectNext(nextStep, getState)) : handleSubmit())}
+  >
+    {nextStep ? nextLabel : submitLabel}
+  </Button>
+);
+
+NextButton.propTypes = {
+  nextStep: PropTypes.oneOfType([PropTypes.string, PropTypes.func, PropTypes.object]),
+  handleSubmit: PropTypes.func.isRequired,
+  submitLabel: PropTypes.node.isRequired,
+  valid: PropTypes.bool,
+  handleNext: PropTypes.func.isRequired,
+  nextLabel: PropTypes.node.isRequired,
+  getState: PropTypes.func.isRequired,
+};
+
+const ReviewStepButtons = ({ disableBack, handlePrev, nextStep, handleNext, buttonLabels: { cancel, submit, back, next } }) => {
+  const formOptions = useFormApi();
+  const { success, error, submitting } = useContext(AddRoleWizardContext);
+  if (success || typeof error === 'undefined' || error || submitting) {
+    return null;
```

Comment:
Wouldn't it be better to return `<Fragment />` here? So we are consistent on return type.

Reply:
It does not really matter. The fragment will only show up in the virtual DOM but the effect will be the same on the actual DOM. And this way there is one less node to traverse in the react DOM with null.